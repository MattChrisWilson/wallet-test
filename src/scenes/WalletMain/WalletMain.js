import React, { Component, Fragment } from 'react';
import Wallet from '../../components/Wallet';
import amendBalance from '../../core/AmendBalance';

// @Flow

type Props = {
  loading: boolean,
  value: number,
  overdraft: boolean,
  setBalance: (data: {newBalance: number, overdraft: boolean}) => void,
};

class WalletMain extends Component<Props> {
  handleOnClick = ({ value, action }) => {
    amendBalance(value, action)
    .then(
      msg => {this.props.setBalance(msg)},
      e => {console.log(e)}
    );
  }

  render() {
    return <Fragment>
      <Wallet 
        loading={this.props.loading} 
        value={this.props.value} 
        onClick={(o) => this.handleOnClick(o)}
      />
      {this.props.overdraft ? <div className={'warning invert rounded'}>You're in your overdraft</div> : ''}
    </Fragment>
  }
}

WalletMain.defaultProps = {
  loading: false,
  value: 0,
  overdraft: false,
}

export default WalletMain;