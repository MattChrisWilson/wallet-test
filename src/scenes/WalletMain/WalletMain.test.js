import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import WalletMain from './WalletMain';

configure({ adapter: new Adapter() });

const setup = (setupProps = {}) => {
	const defaultProps = {
		loading: false,
		value: 0,
    overdraft: false,
	};

	const props = { ...defaultProps, ...setupProps };

	const comp = shallow(
		<WalletMain loading={props.loading} value={props.value} />
	);

	return { props, comp };
}

describe('WalletMain', () => {
  test('Renders without crashing', () => {
    const { comp } = setup();
    expect(comp).toMatchSnapshot();
  });

  test('Renders loading message when loading is true', () => {
    const { comp } = setup({ loading: true });
    expect(comp).toMatchSnapshot();
  });

  test('Renders value when passed', () => {
    const { comp } = setup({ value: 5 });
    expect(comp).toMatchSnapshot();
  });

  test('Renders overdraft message when passed', () => {
    const { comp } = setup({ value: -5, overdraft: true });
    expect(comp).toMatchSnapshot();
  });
});