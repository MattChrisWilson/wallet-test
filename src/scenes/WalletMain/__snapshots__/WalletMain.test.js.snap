// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletMain Renders loading message when loading is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WalletMain
    loading={true}
    overdraft={false}
    value={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Wallet
          loading={true}
          onClick={[Function]}
          value={0}
        />,
        "",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "loading": true,
          "onClick": [Function],
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "",
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Wallet
            loading={true}
            onClick={[Function]}
            value={0}
          />,
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "loading": true,
            "onClick": [Function],
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "",
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`WalletMain Renders overdraft message when passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WalletMain
    loading={false}
    overdraft={false}
    value={-5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Wallet
          loading={false}
          onClick={[Function]}
          value={-5}
        />,
        "",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "loading": false,
          "onClick": [Function],
          "value": -5,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "",
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Wallet
            loading={false}
            onClick={[Function]}
            value={-5}
          />,
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "loading": false,
            "onClick": [Function],
            "value": -5,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "",
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`WalletMain Renders value when passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WalletMain
    loading={false}
    overdraft={false}
    value={5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Wallet
          loading={false}
          onClick={[Function]}
          value={5}
        />,
        "",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "loading": false,
          "onClick": [Function],
          "value": 5,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "",
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Wallet
            loading={false}
            onClick={[Function]}
            value={5}
          />,
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "loading": false,
            "onClick": [Function],
            "value": 5,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "",
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`WalletMain Renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WalletMain
    loading={false}
    overdraft={false}
    value={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Wallet
          loading={false}
          onClick={[Function]}
          value={0}
        />,
        "",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "loading": false,
          "onClick": [Function],
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "",
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Wallet
            loading={false}
            onClick={[Function]}
            value={0}
          />,
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "loading": false,
            "onClick": [Function],
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "",
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
